### PR TITLE
Created a console service

### PR DESCRIPTION
**Pull request by CarlosNihelton**
This service class is intended to work in integration with the previously merged `LocalNamedPipe`, by taking ownership of it and:

- Controling visibility of the console window
- Redirecting the console output to that pipe and restoring from redirection

As before this is not yet integrated to the application main code, so no behavior change is noticeable yet.

### CODE REVIEW

**Review by didrocks — APPROVED** [review]
This looks good to me, code-wise. Small nitpicks to make the comments more future-proof, but really good work :)

File: DistroLauncher-Tests/DistroLauncher-Tests/ConsoleServiceTests.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    }
+    TEST(ConsoleServiceTests, RedirectingAGoodFakeDoesNothingButWontFail)
+    {
+        // This fake pipe simulates the behavior of a healthy pipe by creating a text file and exposing it in 3 ways:
+        // 1. As a Win32 HANDLE
+        // 2. As a File Descriptor
+        // 3. As a File Stream.
+        // 1 and 2 are part of the Pipe API, but 3 is a way to fake the stdout and stderr when calling the console
+        // service redirectConsole method.
+        // By doing so, line 111 effectively redirects this file into itself. This intentionally tricks the console
```
[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply DistroLauncher-Tests/DistroLauncher-Tests/ConsoleServiceTests.cpp:52]: 👍

File: DistroLauncher/console_service.h [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+            return ShowWindow(GetConsoleWindow(), SW_HIDE) == TRUE;
+        }
+
+        bool showConsoleWindow() const
+        {
+            return ShowWindow(GetConsoleWindow(), SW_SHOW) == TRUE;
+        }
+    };
+
+    // Notice that STDIN was never touched. It could be, but it is outside of the scope of what's planned for this
```
[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply DistroLauncher/console_service.h:147]: Agreed.

File: DistroLauncher-Tests/DistroLauncher-Tests/ConsoleServiceTests.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+            {
+                return L"FakePipe";
+            }
+            HANDLE readHandle()
+            {
+                return hFile;
+            }
+            explicit FakePipe(bool inheritRead, bool inheritWrite, const wchar_t* name)
+            {
+                HANDLE handle = CreateFile(L"test-file.txt",      // file name
```
[the remark this answers is not in the document]
Comment (CarlosNihelton) [reply DistroLauncher-Tests/DistroLauncher-Tests/ConsoleServiceTests.cpp]: Yes, and caching its name to ensure removal in the end of the test case.

**Pull request merged**